ogr/ogrsf_frmts/sqlite/ogrsqlitedatasource.cpp

```diff
@@ -284,7 +284,10 @@ std::vector<std::string> OGRSQLiteDataSource::GetRelationshipNames(
 
 {
     if (!m_bHasPopulatedRelationships)
-        LoadRelationshipsFromForeignKeys();
+    {
+        m_osMapRelationships.clear();
+        LoadRelationshipsFromForeignKeys({});
+    }
```

Comment:
A possibly cleaner solution would be to have a *virtual* LoadRelationships() method in OGRSQLiteBaseDataSource that would do:
```
void OGRSQLiteDataSource::LoadRelationships()
{
        m_osMapRelationships.clear();
        LoadRelationshipsFromForeignKeys({});
        m_bHasPopulatedRelationships = true;
}
```
and call it from there
and have the LoadRelationships() method of GPKG override it
and remove the ``m_bHasPopulatedRelationships = true;`` statement from ogrsqlitedatasource.cpp:831